Add Firefox Privacy Hub pages (Fixes #7766, #7709)

alexgibson opened the pull request:
## Description
- Adds pages at `/firefox/privacy/` and `/firefox/privacy/products/`

This page needs to go to production on the **21st October (AM PST)**.

## Issue / Bugzilla link
#7709

## Testing
Demo:
- https://www-demo1.allizom.org/en-US/firefox/privacy/
- https://www-demo1.allizom.org/en-US/firefox/privacy/products/

Successful test run: https://ci.vpn1.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/372/pipeline

alexgibson commented:
I think this is now close enough to be ready for code review (we're just pending a couple of final minor tweaks).

stephaniehobson commented on media/css/firefox/privacy/products.scss:
```diff
@@ -0,0 +1,298 @@
+
+// Underline active link in sub navigation.
+.privacy-products .c-sub-navigation-item.privacy-products > a {
+    text-decoration: underline;
```
I'm worried it will be confusing for users to have something that is not a link underlined. Do you think we could move back towards the original design direction for the subnav?

<img width="355" alt="Screen Shot 2019-10-03 at 14 57 15" src="https://user-images.githubusercontent.com/854701/66167154-38d57080-e5ee-11e9-9566-7fc1dcbe7d52.png">

alexgibson replied:
I'm worried that what's in the design is even more confusing, since the bold link is indistinguishable (in it's meaning) when comparing to the (already bold) links in the main navigation. Jeniffer said an underline here is OK, but let me circle back and see if there's anything better we can do compared to either solution.

stephaniehobson commented on media/css/firefox/privacy/products.scss:
```diff
@@ -0,0 +1,298 @@
+.privacy-products-etp {
+    background-color: $color-off-black;
+
+    &> .mzp-l-content {
```
Nit: space between `&>`

stephaniehobson commented on media/css/firefox/privacy/products.scss:
```diff
@@ -0,0 +1,298 @@
+    }
+
+    .mzp-c-card-feature-media img {
+        max-width: 320px;
```
I do this (restrict feature card image width on small screen sizes) a lot. Do you think it's worth adding it to the base Protocol styles?

alexgibson replied:
I can see quite a few bedrock pages where this isn't the case, or it at least varies in size.

stephaniehobson commented on media/css/firefox/privacy/products.scss:
```diff
@@ -0,0 +1,298 @@
+}
+
+.privacy-products-lockwise .mzp-c-card-feature-title {
+    @include at2x('/media/protocol/img/logos/firefox/lockwise/logo-word-hor-md.png', auto, 40px);
```
I think you can use the `-sm` size for the monitor and send like you do the monitor one.

stephaniehobson commented:
It feels a little weird to see the sign in form as a user who is already signed in. Is that the intended behaviour?

stephaniehobson commented:
iOS is looking good. I just have one more file to look at for my review, but it will have to wait for tomorrow.

alexgibson commented on bedrock/firefox/templates/firefox/privacy/base.html:
```diff
@@ -0,0 +1,24 @@
+
+{% add_lang_files "firefox/privacy-hub" %}
+
+{% extends "firefox/base/base-protocol.html" %}
```
Good suggestion, I'll add the master brand ones for the time being (seeing as we have those). We can always update later too.

alexgibson commented on bedrock/firefox/templates/firefox/privacy/products.html:
```diff
@@ -0,0 +1,187 @@
+    <ul class="mzp-l-content mzp-l-card-third mzp-t-dark">
+      {{ picto_card(title=_('Invisible to the top trackers'), desc=_('Meet four of the most common categories of trackers — who won’t meet you.'), class='invisible js-open-about-protections') }}
+      {{ picto_card(title=_('Always in your control'), desc=_('Want to customize what gets blocked? Your settings are only one click away.'), class='control js-open-about-protections') }}
+      {{ picto_card(title=_('Protection beyond tracking'), desc=_('If you have a Firefox account, you can also see how we’re helping you protect your personal info and passwords.'), class='protection js-open-about-protections') }}
```
This is a last minute request from the main stakeholder, so it needs to stay. They feel these headings are more eye catching than the CTA links (which are on the small side), 

I agree on the semantics though, so I'll see if we can make just the headings clickable instead of the combined copy.

alexgibson commented:
> It feels a little weird to see the sign in form as a user who is already signed in. Is that the intended behaviour?

The form is the last thing on the page, and very much a secondary CTA, so there's no conditional content for it. From a 'feeling weird' point of view, we do already do this for all non-Firefox browsers (since trailhead) on pages where this form apears, so I don't think it's too strange any more. There's also an "Already have an account? Sign In" link.

alexgibson commented:
Thanks for the review @stephaniehobson - I think I've addressed all feedback so far, as well as incorporated some other last minute tweaks from other stakeholders.

For the sub navigation styling, I've opened a conversation in the issue with Jennifer: https://github.com/mozilla/bedrock/issues/7765#issuecomment-538328080

stephaniehobson commented on media/js/firefox/privacy/products.js:
```diff
@@ -0,0 +1,56 @@
+                    pictoCardTitles[j].setAttribute('role', 'link');
+                    pictoCardTitles[j].setAttribute('tabindex', 0);
+                    pictoCardTitles[j].addEventListener('click', handleOpenProtectionReport, false);
+                    pictoCardTitles[j].addEventListener('keydown', function(e) {
```
This need to handle touch events too (the click is not being triggered on my Surface).

stephaniehobson replied:
Yup, I just didn't have the demo configured on the Surface.

alexgibson commented:
> This need to handle touch events too (the click is not being triggered on my Surface).

I don't have one of these devices to test on, but I'm not sure there should be a need for a touch event when there's alredy a click event bound. The click should only work when UITour is enabled - is there a chance your surface isn't set up?

stephaniehobson commented on bedrock/base/templates/includes/protocol/footer/firefox-new.html:
```diff
@@ -13,7 +13,7 @@ <h5 class="mzp-c-footer-heading">
           </h5>
           <ul class="mzp-c-footer-list">
-            <li><a href="{{ url('privacy.notices.firefox') }}" data-link-type="footer" data-link-name="Privacy">{{ _('Privacy') }}</a></li>
+            <li><a href="{{ url('firefox.privacy.index') }}" data-link-type="footer" data-link-name="Privacy">{{ _('Privacy') }}</a></li>
```
Thank you!

stephaniehobson commented on bedrock/firefox/templates/firefox/privacy/includes/subnav.html:
```diff
@@ -7,8 +7,20 @@
-        <li class="c-sub-navigation-item privacy-promise"><a href="{{ url('firefox.privacy.promise') }}">{{ _('Our Promise') }}</a></li>
-        <li class="c-sub-navigation-item privacy-products"><a href="{{ url('firefox.privacy.products') }}">{{ _('Our Products') }}</a></li>
+        <li class="c-sub-navigation-item privacy-promise">
+          {% if active == 'promise' %}
```
🔥

stephaniehobson commented on media/js/firefox/privacy/products.js:
```diff
@@ -26,25 +26,26 @@
+
+                for (var j = 0; j < pictoCardTitles.length; j++) {
+                    // Make picto card titles click and keyboard accessible.
+                    pictoCardTitles[j].setAttribute('role', 'link');
```
Thanks for restricting this to the headings.

stephaniehobson commented:
R+ 🔒 

Merge at will. Left it open in case there is still stakeholder feedback coming in.

alexgibson commented:
Thanks @stephaniehobson! This can't go our until the 21st, so I'm going to keep it here and rebase until then. I'll let you know if any changes creep in. Thanks again for the great review!

alexgibson commented:
@stephaniehobson I pushed 2 small updates to this PR:

1. Change lockwise link to point to /firefox/lockwise.
2. Add FxA sign-in link JS from the /accounts page.

stephaniehobson commented:
@alexgibson the purple box on the new /firefox homepage can be updated to link to the privacy landing page now that #7776 is merged.

stephaniehobson commented:
This still looks good ⭐️
